pkg/query-service/app/integrations/messagingQueues/kafka/translator.go

```diff
@@ -35,6 +35,139 @@ func BuildQueryRangeParams(messagingQueue *MessagingQueue, queryContext string)
 	return queryRangeParams, nil
 }
 
+func buildClickHouseQueryNetwork(messagingQueue *MessagingQueue, queueType string) (*v3.ClickHouseQuery, error) {
+	start := messagingQueue.Start
+	end := messagingQueue.End
+	consumerGroup, ok := messagingQueue.Variables["consumer_group"]
+	if !ok {
+		return nil, fmt.Errorf("consumer_group not found in the request")
+	}
+
+	query := generateNetworkLatencyThroughputSQL(start, end, consumerGroup, queueType)
+
+	return &v3.ClickHouseQuery{
+		Query: query,
+	}, nil
+}
+
+func buildBuilderQueriesNetwork(attributeCache []Clients) (map[string]*v3.BuilderQuery, error) {
+	bq := make(map[string]*v3.BuilderQuery)
+
+	for i, instanceInfo := range attributeCache {
+		queryName := fmt.Sprintf("latency_%d", i)
+		chq := &v3.BuilderQuery{
+			QueryName:    queryName,
+			StepInterval: defaultStepInterval,
```

Comment:
Please use [`MinAllowedStepInterval`](https://github.com/SigNoz/signoz/blob/c2f607ab6bc1766368b809df841be1a7687d5447/pkg/query-service/common/query_range.go#L45). It will adjust the interval based on the selected time range.

Reply:
done